[Florentiana Yuwono] iP

florentianayuwono opened the pull request:
# DukeFloren

> "Even if you are not productive, at least try to look like it." - Duke [(source)](https://dansilvestre.com/productivity-quotes/)

DukeFloren helps you to look productive in front of others, even though you are really not. Why? Cuz it's

- text-based
- look super cool
- ~~FAST~~ *PRETTY* FAST

Aight. Now, all you need to do is,

1. download it from [here](https://github.com/florentianayuwono/ip)
2. run it in your local machine
3. add your tasks
4. and pretend to look productive! 🚀🦸‍♂️

The best part is? It's **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

florentianayuwono committed:
Level-1

florentianayuwono committed:
Level-2: Add and list commands

florentianayuwono committed:
Level-1 and Level-2 files added

florentianayuwono committed:
Level-3 Mark as Done

florentianayuwono committed:
Level 4 completed

florentianayuwono committed:
create exception

florentianayuwono committed:
add more exceptions

florentianayuwono committed:
add delete functionality

florentianayuwono committed:
add input text

florentianayuwono committed:
run automated test

florentianayuwono committed:
add funcitionalities

florentianayuwono committed:
solve automated text issue

florentianayuwono committed:
implement basic save list to file

florentianayuwono committed:
add datetime basic functionality

florentianayuwono committed:
add modifying task file functionality

florentianayuwono committed:
fix time parse error

florentianayuwono committed:
fix merge problems

florentianayuwono committed:
fix bug adding list to tasklist

florentianayuwono committed:
add OOP skeleton

florentianayuwono committed:
add class Ui and Storage

florentianayuwono committed:
add class Storage and Parser

florentianayuwono committed:
add Command class

florentianayuwono committed:
close scanner in Storage class

florentianayuwono committed:
finish all functionality for OOP

florentianayuwono committed:
organize into packages

florentianayuwono committed:
fix bug mark and unmark

florentianayuwono committed:
Merge branch 'add-gradle-support' of https://github.com/florentianayuwono/ip into A-Gradle

florentianayuwono committed:
add javadoc to Duke.java

florentianayuwono committed:
fix build issue

florentianayuwono committed:
add JUnit tests

florentianayuwono committed:
complete javadoc for all classes

florentianayuwono committed:
fix style violations

florentianayuwono committed:
Merge tag 'A-JavaDoc'

florentianayuwono committed:
remove unused imports

florentianayuwono committed:
add find functionality

heyzec commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    public void run() {
+        ui.showWelcomeMessage();
+        boolean isExit = false;
```
Perhaps it's clearer to name this `shouldExit`?

florentianayuwono replied:
Oh yes, I think it is more intuitive to name it as `shouldExit`. Thanks!

heyzec commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,91 @@
+
+        switch (commandWord) {
+
+            case AddCommand.ADD_TODO:
```
Our guidelines says `case` should be on the same indentation as `switch`.

florentianayuwono replied:
You are right! Thanks for reminding me :)

heyzec commented on src/test/java/duke/commands/MarkCommandTest.java:
```diff
@@ -0,0 +1,19 @@
+package duke.commands;
+
+import duke.exception.DukeException;
+import duke.task.ToDo;
```
This import can be safely removed.

florentianayuwono replied:
Done, thanks!

heyzec commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,17 @@
+import duke.commands.AddCommand;
+import duke.exception.DukeException;
+import duke.task.ToDo;
+import org.junit.jupiter.api.Test;
```
Some imports can be cleaned here.

florentianayuwono replied:
Cleaned! Thanks for reminding :)

heyzec reviewed (commented):
Great work and design!

jialatteo commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Represents an executable command.
+ */
+public abstract class Command {
+    /**
+     * Executes the command and returns the result.
+     */
```
Good implementation of Command abstract class! 👍

florentianayuwono replied:
Thank you!

jialatteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,122 @@
+                String taskStatus = commands.substring(4, 5);
+                String taskDescription = commands.substring(7);
+
+                switch (taskType) {
+                    case "T":
+                        Task toDo = new ToDo(taskDescription);
+                        if (taskStatus.equals("X")) {
+                            toDo.markTaskAsDone();
+                        }
+                        tasks.addTask(toDo);
+                        break;
+                    case "D":
+                        String[] deadlineDescription = taskDescription.split("by:");
+                        String deadlineName = deadlineDescription[0]
+                                .substring(0, deadlineDescription[0].length() - 1)
+                                .trim();
+                        String by = deadlineDescription[1].substring(1, deadlineDescription[1].length() - 1).trim();
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate date = LocalDate.parse(by, formatter);
+                        Task deadline = new Deadline(deadlineName, date);
+                        if (taskStatus.equals("X")) {
+                            deadline.markTaskAsDone();
+                        }
+                        tasks.addTask(deadline);
+                        break;
+                    case "E":
+                        String[] eventDescription = taskDescription.split("at:");
+                        String eventName = eventDescription[0]
+                                .substring(0, eventDescription[0].length() - 1)
+                                .trim();
+                        String at = eventDescription[1].substring(0, eventDescription[1].length() - 1).trim();
+                        Task event = new Event(eventName, at);
+                        if (taskStatus.equals("X")) {
+                            event.markTaskAsDone();
+                        }
+                        tasks.addTask(event);
+                        break;
+                }
```
Perhaps can remove indentations for case clauses?

florentianayuwono replied:
True! Thank you for reminding :D

jialatteo commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,44 @@
+            + "%1$s" + "\n"
+            + "You have " + "%2$s" + " tasks in the list.\n";
+
+    public Task toAdd;
```
Maybe can set to private instead?

florentianayuwono replied:
True. Implemented already, thanks!

jialatteo commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,122 @@
+                        }
+                        tasks.addTask(toDo);
+                        break;
+                    case "D":
+                        String[] deadlineDescription = taskDescription.split("by:");
+                        String deadlineName = deadlineDescription[0]
+                                .substring(0, deadlineDescription[0].length() - 1)
+                                .trim();
+                        String by = deadlineDescription[1].substring(1, deadlineDescription[1].length() - 1).trim();
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate date = LocalDate.parse(by, formatter);
+                        Task deadline = new Deadline(deadlineName, date);
+                        if (taskStatus.equals("X")) {
+                            deadline.markTaskAsDone();
+                        }
+                        tasks.addTask(deadline);
+                        break;
+                    case "E":
+                        String[] eventDescription = taskDescription.split("at:");
+                        String eventName = eventDescription[0]
+                                .substring(0, eventDescription[0].length() - 1)
+                                .trim();
+                        String at = eventDescription[1].substring(0, eventDescription[1].length() - 1).trim();
+                        Task event = new Event(eventName, at);
+                        if (taskStatus.equals("X")) {
+                            event.markTaskAsDone();
+                        }
+                        tasks.addTask(event);
+                        break;
```
Maybe can consider to have a function to parse Events and Deadline as there is a bit of code duplication here?

florentianayuwono replied:
Good suggestion!

jialatteo reviewed (commented):
Overall great work with the chatbot! 😁

florentianayuwono committed:
add config checkstyle file

florentianayuwono committed:
set base for Duke GUI

florentianayuwono committed:
complete GUI functionality

florentianayuwono committed:
Add CheckStyle and fix coding standard violations

florentianayuwono committed:
fix merge conflicts

florentianayuwono committed:
add java varargs

florentianayuwono committed:
Add assertions to several methods

There are several methods that need assert in order to make sure the value inputted is valid.

For example, to assert that the input value is not null or negative integer.

florentianayuwono committed:
Improve code quality

Practice better naming convention for class fields.

Reduce code duplication and method length.

florentianayuwono committed:
Merge pull request #1 from florentianayuwono/branch-A-CodeQuality

Improve code quality

florentianayuwono committed:
Merge branch 'master' into branch-A-Assertions

florentianayuwono committed:
Merge pull request #2 from florentianayuwono/branch-A-Assertions

Add assertions to several methods

florentianayuwono committed:
Set up CI

florentianayuwono committed:
add task duplication detection

florentianayuwono committed:
fix style error

florentianayuwono committed:
fix style error

florentianayuwono committed:
add product image

florentianayuwono committed:
Update README.md

florentianayuwono committed:
Update README.md

florentianayuwono committed:
Update README.md

florentianayuwono committed:
Fix header comment styling

florentianayuwono committed:
Merge branch 'master' of https://github.com/florentianayuwono/ip

florentianayuwono committed:
Make dialog box responsive to wrap long text

florentianayuwono committed:
Change greet message

florentianayuwono committed:
Add error handling for unknown command

florentianayuwono committed:
Add private parseToDo method

florentianayuwono committed:
Add private parse task methods

florentianayuwono committed:
Add more error handling to each parse methods

florentianayuwono committed:
Show error message if not task found

florentianayuwono committed:
Add private load task methods

florentianayuwono committed:
Enhance find task functionality

florentianayuwono committed:
Enhance detect duplicate functionality

florentianayuwono committed:
Fix project errors

florentianayuwono committed:
Fix JavaFX version

florentianayuwono committed:
change mainClassName to Launcher
